pkg/plugins/resources/postgres/pgx_store.go

```diff
@@ -0,0 +1,382 @@
+	}
+	if err != nil {
+		return errors.Wrapf(err, "failed to execute query: %s", statement)
+	}
+
+	if err := core_model.FromJSON([]byte(spec), resource.GetSpec()); err != nil {
+		return errors.Wrap(err, "failed to convert json to spec")
+	}
+
+	meta := &resourceMetaObject{
+		Name:             opts.Name,
+		Mesh:             opts.Mesh,
+		Version:          strconv.Itoa(version),
+		CreationTime:     creationTime.Local(),
+		ModificationTime: modificationTime.Local(),
+	}
+	resource.SetMeta(meta)
+
+	if opts.Version != "" && resource.GetMeta().GetVersion() != opts.Version {
+		return store.ErrorResourcePreconditionFailed(resource.Descriptor().Name, opts.Name, opts.Mesh)
+	}
+	return nil
+}
+
+func (r *pgxResourceStore) List(ctx context.Context, resources core_model.ResourceList, args ...store.ListOptionsFunc) error {
+	opts := store.NewListOptions(args...)
+
+	statement := `SELECT name, mesh, spec, version, creation_time, modification_time FROM resources WHERE type=$1`
+	var statementArgs []interface{}
+	statementArgs = append(statementArgs, resources.GetItemType())
+	argsIndex := 1
+	if opts.Mesh != "" {
+		argsIndex++
+		statement += fmt.Sprintf(" AND mesh=$%d", argsIndex)
+		statementArgs = append(statementArgs, opts.Mesh)
+	}
+	if opts.NameContains != "" {
+		argsIndex++
+		statement += fmt.Sprintf(" AND name LIKE $%d", argsIndex)
+		statementArgs = append(statementArgs, "%"+opts.NameContains+"%")
+	}
+	statement += " ORDER BY name, mesh"
+
+	rows, err := r.pool.Query(ctx, statement, statementArgs...)
+	if err != nil {
+		return errors.Wrapf(err, "failed to execute query: %s", statement)
+	}
+	defer rows.Close()
+
+	total := 0
+	for rows.Next() {
+		item, err := rowToItem(resources, rows)
+		if err != nil {
+			return err
+		}
+		if err := resources.AddItem(item); err != nil {
+			return err
+		}
+		total++
+	}
+
+	resources.GetPagination().SetTotal(uint32(total))
+	return nil
+}
+
+func rowToItem(resources core_model.ResourceList, rows pgx.Rows) (core_model.Resource, error) {
+	var name, mesh, spec string
+	var version int
+	var creationTime, modificationTime time.Time
+	if err := rows.Scan(&name, &mesh, &spec, &version, &creationTime, &modificationTime); err != nil {
+		return nil, errors.Wrap(err, "failed to retrieve elements from query")
+	}
+
+	item := resources.NewItem()
+	if err := core_model.FromJSON([]byte(spec), item.GetSpec()); err != nil {
+		return nil, errors.Wrap(err, "failed to convert json to spec")
+	}
+
+	meta := &resourceMetaObject{
+		Name:             name,
+		Mesh:             mesh,
+		Version:          strconv.Itoa(version),
+		CreationTime:     creationTime.Local(),
+		ModificationTime: modificationTime.Local(),
+	}
+	item.SetMeta(meta)
+
+	return item, nil
+}
+
+func (r *pgxResourceStore) Close() error {
+	r.pool.Close() // does not work, at least on my machine
+	return nil
+}
+
+func registerMetrics(metrics core_metrics.Metrics, pool *pgxpool.Pool) error {
```

Comment:
For the metrics I tried to mirror what's already in `pq` and there are some additional metrics in `pgx`. Please view the diff between pgx_store and pq_store using:

```
git diff --no-index pkg/plugins/resources/postgres/pq_store.go  pkg/plugins/resources/postgres/pgx_store.go
```

or this online preview: https://htmlpreview.github.io/?https://gist.githubusercontent.com/slonka/4a54b687151888790a5447a35e62c719/raw/758dd8fc124e4010d751c39f537d064e136ea313/pq_pgx_store_difference.html